pandas/core/arrays/datetimelike.py

```diff
@@ -713,7 +713,7 @@ def map(self, mapper):
         # Else, just rewrite _map_infer_values to do the right thing.
         from pandas import Index
 
-        return Index(self).map(mapper).array
+        return Index(self).map(mapper)
```

Comment:
I suspect this isn't the fix that we would want although test are passing. We want ExtensionArrays methods to return ExtensionArrays and not Indexes. Could you investigate another fix (maybe in apply?)

Reply:
cc @jbrockmendel , as there was further discussion in the issue regarding a proper patch

Reply:
I've reverted my changes and move the fix down to apply